src/extjson/mod.rs

```diff
@@ -0,0 +1,81 @@
+//! Deserialization and serialization of [MongoDB Extended JSON v2](https://docs.mongodb.com/manual/reference/mongodb-extended-json/)
```

Comment:
this module is also largely empty, but it could export the various functions we eventually define in the `ser` and `de` submodules like we do for BSON.
e.g.
```rust
// in `extjson/mod.rs`
pub use de::from_value;
pub use ser::to_value;

// in users' code
let ext_json = extjson::to_value(5).unwrap();
let x: i32 = extjson::from_value(json!({"$numberInt: "5" })).unwrap();
```
The way I see it, extJSON is basically its own serialization format, so it makes sense to have a similar module hierarchy and functionality for it.